src/cmd/ksh93/sh/path.c

```diff
@@ -393,7 +393,9 @@ Pathcomp_t *path_nextcomp(Shell_t *shp, Pathcomp_t *pp, const char *name, Pathco
 
 static_fn Pathcomp_t *defpath_init(Shell_t *shp) {
     if (!std_path) {
-        std_path = astconf("PATH", NULL, NULL);
+        // Value returned by astconf() is short lived, duplicate the string.
+        // https://github.com/att/ast/issues/959
+        std_path = strdup(astconf("PATH", NULL, NULL));
```

Comment:
Shouldn't the original, non strdup, assignment be done then this becomes:

```
if (std_path) {
    std_path = strdup(std_path);
    assert(std_path);
} else {
    std_path = e_defpath;
}
```

Reply:
Fixed.